acceptance/analyzer_test.go

```diff
@@ -145,34 +127,57 @@ func setupAnalyzeFixtures(t *testing.T) analyzeFixtures {
 
 	// Registry
 
+	// setup read and write images
 	someAppName := "some-app-image-" + h.RandString(10)
 	fixtures.authRegAppImage, fixtures.authRegAuthConfig = buildAuthRegistryImage(
 		t,
 		someAppName,
 		filepath.Join("testdata", "analyzer", "app-image"),
+		testRegistry,
 		"--build-arg", "fromImage="+containerBaseImage,
 		"--build-arg", "metadata="+appMeta,
 	)
-	fixtures.readOnlyRegAppImage = readOnlyRegistry.RepoName(someAppName)
+	testRegistry.Add(someAppName, ih.NewImagePrivileges(readWriteImage))
 
 	someOtherAppName := "some-other-app-image-" + h.RandString(10)
 	fixtures.authRegOtherAppImage, _ = buildAuthRegistryImage(
 		t,
 		someOtherAppName,
 		filepath.Join("testdata", "analyzer", "app-image"),
+		testRegistry,
 		"--build-arg", "fromImage="+containerBaseImage,
 		"--build-arg", "metadata="+appMeta,
 	)
+	testRegistry.Add(someOtherAppName, ih.NewImagePrivileges(readWriteImage))
 
-	someCacheName := "some-cache-image-" + h.RandString(10)
+	// setup read only images
+	someReadOnlyAppName := "some-readonly-app-image-" + h.RandString(10)
+	fixtures.readOnlyRegAppImage, _ = buildAuthRegistryImage(
+		t,
+		someReadOnlyAppName,
+		filepath.Join("testdata", "analyzer", "app-image"),
+		testRegistry,
+		"--build-arg", "fromImage="+containerBaseImage,
+		"--build-arg", "metadata="+appMeta,
+	)
+	testRegistry.Add(someReadOnlyAppName, ih.NewImagePrivileges(onlyReadImage))
+
+	someCacheName := "some-cache-image"
 	fixtures.authRegCacheImage, _ = buildAuthRegistryImage(
 		t,
 		someCacheName,
 		filepath.Join("testdata", "analyzer", "cache-image"),
+		testRegistry,
 		"--build-arg", "fromImage="+containerBaseImage,
 		"--build-arg", "metadata="+cacheMeta,
 	)
-	fixtures.readOnlyRegCacheImage = readOnlyRegistry.RepoName(someCacheName)
+	testRegistry.Add(someCacheName, ih.NewImagePrivileges(onlyReadImage))
```

Comment:
I'm confused how this is working. The tests should fail if the cache image is read only.

Reply:
There are a couple of tests actually verifying the read/write error is thrown when a read-only cache image is used. But, there are already two test cases using `fixtures.readOnlyRegCacheImage`, remember I am not adding or modifying the test cases, I am just changing the reference to the image they are using. Those test cases are validating if `cache metadata is restore`